flow/envs/multiagent/i210.py

```diff
@@ -109,38 +140,48 @@ def action_space(self):
     def _apply_rl_actions(self, rl_actions):
         """See class definition."""
         # in the warmup steps, rl_actions is None
+        id_list = []
+        accel_list = []
         if rl_actions:
+            t = time()
             for rl_id, actions in rl_actions.items():
                 accel = actions[0]
 
                 # lane_change_softmax = np.exp(actions[1:4])
                 # lane_change_softmax /= np.sum(lane_change_softmax)
                 # lane_change_action = np.random.choice([-1, 0, 1],
                 #                                       p=lane_change_softmax)
-
-                self.k.vehicle.apply_acceleration(rl_id, accel)
-                # self.k.vehicle.apply_lane_change(rl_id, lane_change_action)
+                id_list.append(rl_id)
+                accel_list.append(accel)
+            self.k.vehicle.apply_acceleration(id_list, accel_list)
+            # self.k.vehicle.apply_lane_change(rl_id, lane_change_action)
+            # print('time to apply actions is ', time() - t)
 
     def get_state(self):
         """See class definition."""
         if self.lead_obs:
             veh_info = {}
             for rl_id in self.k.vehicle.get_rl_ids():
-                speed = self.k.vehicle.get_speed(rl_id)
-                lead_id = self.k.vehicle.get_leader(rl_id)
-                if lead_id in ["", None]:
-                    # in case leader is not visible
-                    lead_speed = SPEED_SCALE
-                    headway = HEADWAY_SCALE
-                else:
-                    lead_speed = self.k.vehicle.get_speed(lead_id)
-                    headway = self.k.vehicle.get_headway(rl_id)
-                veh_info.update({rl_id: np.array([speed / SPEED_SCALE, headway / HEADWAY_SCALE,
-                                                  lead_speed / SPEED_SCALE])})
+                if (self.control_range and self.k.vehicle.get_x_by_id(rl_id) < self.control_range[1] \
```

Comment:
I'd suggest making a in_control_range function to simplify this here and in the other places with the same logic (I think mostly in this file)